Src/Couchbase.Linq/Versioning/DefaultVersionProvider.cs

```diff
@@ -0,0 +1,200 @@
+                // Only check one cluster at a time, this prevents multiple lookups during bootstrap
+                lock (_lock)
+                {
+                    // In case the version was received while we were waiting for the lock, check for it again
+                    version = CacheLookup(servers);
+                    if (version != null)
+                    {
+                        return version;
+                    }
+
+                    foreach (var server in Shuffle(servers))
+                    {
+                        try
+                        {
+                            var config = DownloadConfig(server).Result;
+
+                            version = ExtractVersion(config);
+                            if (version != null)
+                            {
+                                CacheStore(servers, version);
+
+                                return version;
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            Log.ErrorFormat("Unable to load config from {0}", e, server);
+                        }
+                    }
+
+                    // No version information could be loaded from any node
+                    var fallbackVersion = new Version(4, 0, 0);
+                    CacheStore(servers, fallbackVersion);
+                    return fallbackVersion;
+                }
+            }
+            finally
+            {
+                if (contextCache != null)
+                {
+                    SynchronizationContext.SetSynchronizationContext(contextCache);
+                }
+            }
+        }
+
+        internal virtual async Task<Bootstrap> DownloadConfig(Uri uri)
+        {
+            try
+            {
+                using (var handler = new WebRequestHandler())
+                {
+                    handler.ServerCertificateValidationCallback = ServerCertificateValidationCallback;
+
+                    using (var httpClient = new HttpClient(handler))
+                    {
+                        httpClient.Timeout = TimeSpan.FromSeconds(5);
+
+                        var response = await httpClient.GetAsync(uri);
+
+                        response.EnsureSuccessStatusCode();
+
+                        return JsonConvert.DeserializeObject<Bootstrap>(await response.Content.ReadAsStringAsync());
+                    }
+                }
+            }
+            catch (AggregateException ex)
+            {
+                // Unwrap the aggregate exception
+                throw new HttpRequestException(ex.InnerException.Message, ex.InnerException);
+            }
+        }
+
+        private static bool ServerCertificateValidationCallback(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
+        {
+            Log.Info(m => m("Validating certificate: {0}", sslPolicyErrors));
+            return true;
```

Comment:
Yeah, you're right.  I just copied it from the main SDK at https://github.com/couchbase/couchbase-net-client/blob/master/Src/Couchbase/Configuration/Server/Providers/Streaming/HttpServerConfig.cs#L153.

Is that something we need to fix there as well, or was it intentional?

Brant